File: SDKs/amazonaws.com/securityhub/2018-10-26/typescript/src/sdk/models/shared/awscorsconfiguration.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// AwsCorsConfiguration
/** 
 * Contains the cross-origin resource sharing (CORS) configuration for the API. CORS is only supported for HTTP APIs.
**/
export class AwsCorsConfiguration extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=AllowCredentials" })
  allowCredentials?: boolean;

  @SpeakeasyMetadata({ data: "json, name=AllowHeaders" })
  allowHeaders?: string[];

  @SpeakeasyMetadata({ data: "json, name=AllowMethods" })
  allowMethods?: string[];

  @SpeakeasyMetadata({ data: "json, name=AllowOrigins" })
  allowOrigins?: string[];

  @SpeakeasyMetadata({ data: "json, name=ExposeHeaders" })
  exposeHeaders?: string[];

  @SpeakeasyMetadata({ data: "json, name=MaxAge" })
  maxAge?: number;
}

File: SDKs/amazonaws.com/securityhub/2018-10-26/python/src/sdk/models/shared/awscorsconfiguration.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclass
class AwsCorsConfiguration:
    r"""AwsCorsConfiguration
    Contains the cross-origin resource sharing (CORS) configuration for the API. CORS is only supported for HTTP APIs.
    """
    
    allow_credentials: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AllowCredentials') }})
    allow_headers: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AllowHeaders') }})
    allow_methods: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AllowMethods') }})
    allow_origins: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AllowOrigins') }})
    expose_headers: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ExposeHeaders') }})
    max_age: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MaxAge') }})
    
